solana/modules/nft_bridge/program/src/api/transfer.rs

```diff
@@ -180,8 +179,10 @@ pub fn transfer_native(
     );
     invoke(&transfer_ix, ctx.accounts)?;
 
-    let metadata: Metadata =
-        Metadata::from_account_info(accs.spl_metadata.info()).ok_or(InvalidMetadata)?;
+    let metadata = {
+        let mut data: &[u8] = &accs.spl_metadata.info().try_borrow_data()?;
+        mpl_token_metadata::utils::meta_deser_unchecked(&mut data).map_err(|_| InvalidMetadata)?
+    };
```

Comment:
The previous implementation of `from_account_info` https://github.com/wormhole-foundation/wormhole/blob/99235bc08d0dd2b18ebae79354c2ba671166238c/solana/modules/token_bridge/token-metadata/src/state.rs#L112C1-L114 seems to perform the same as this new version (with no additional checks). It does seem like the new version of `from_account_info` does extra checks, so I'm also curious why we decided not to perform those.

Reply:
I've added a method called `deserialize_and_verify_metadata` for both Token Bridge and NFT Bridge programs. Like how `from_account_info` performed a few checks before returning the deserialized `Metadata` account, this condenses all of the account checks, including verifying pda, checking the account owner and checking whether data is empty.

There may be some redundancy with some checks, especially with the `update_accounts` method. But this may be what we want since not every method that requires this `Metadata` account interacts w/ Metaplex's Token Metadata program.

How does it look @kcsongor @felix-jump?